src/core/lib/math/discretegaussiangenerator.cpp

```diff
@@ -47,6 +47,15 @@ DiscreteGaussianGeneratorImpl<VecType>::DiscreteGaussianGeneratorImpl(double std
     SetStd(std);
 }
 
+template <typename VecType>
+bool DiscreteGaussianGeneratorImpl<VecType>::IsInitialized() const {
+    if (abs(m_std - 1) < 1e-9) {
```

Comment:
1.  i am not sure if you want to get the absolute value here. do you expect m_std to be less then 1 or even negative?
2. also for this kind of functions the code should check the true condition first and return false for anything else:
**return ((m_std - 1) > 1e-9) ? true : false;**